Dev Client on connected Could not initialize class

Client ran into exception after connecting to local dev server.

Checked out Commit: d745141b7bf09fbfe5f3cd92ece615af80e7d988

<details>
<summary>Client Log</summary>

```
INFO  2022-04-19 12:32:44,375 Starting MAGE client version 1.4.50-V2 (build: runtime)                                    =>[main] MageFrame.main 
INFO  2022-04-19 12:32:44,377 Logging level: INFO                                                                        =>[main] MageFrame.main 
INFO  2022-04-19 12:32:44,377 Default charset: UTF-8                                                                     =>[main] MageFrame.main 
INFO  2022-04-19 12:32:45,326 Using GUI theme: Default                                                                   =>[AWT-EventQueue-0] PreferencesDialog.getCurrentTheme 
INFO  2022-04-19 12:32:45,337 Loading database...                                                                        =>[AWT-EventQueue-0] RepositoryUtil.bootstrapLocalDb 
INFO  2022-04-19 12:32:45,626 DB: creating cards database (it can take few minutes)...                                   =>[AWT-EventQueue-0] MageFrame.<init> 
INFO  2022-04-19 12:32:45,632 Done.                                                                                      =>[AWT-EventQueue-0] MageFrame.<init> 
INFO  2022-04-19 12:32:45,632 Images: search broken files...                                                             =>[AWT-EventQueue-0] MageFrame.<init> 
INFO  2022-04-19 12:32:45,872 Loading symbols...                                                                         =>[AWT-EventQueue-0] ManaSymbols.loadImages 
INFO  2022-04-19 12:32:46,030 Loading plugins...                                                                         =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-04-19 12:32:46,061 [Game counter plugin, version 0.1] has been loaded.                                        =>[AWT-EventQueue-0] CounterPluginImpl.newPlugin 
INFO  2022-04-19 12:32:46,068 [Card plugin, version 0.7] has been loaded.                                                =>[AWT-EventQueue-0] CardPluginImpl.newPlugin 
INFO  2022-04-19 12:32:46,069 [Theme plugin, version 0.5] has been loaded.                                               =>[AWT-EventQueue-0] ThemePluginImpl.newPlugin 
INFO  2022-04-19 12:32:46,069 Done.                                                                                      =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-04-19 12:32:46,602 Client start up time: 2 seconds                                                            =>[AWT-EventQueue-0] MageFrame.lambda$new$2 
INFO  2022-04-19 12:32:48,698 Connect: connecting to server localhost:17171                                              =>[SwingWorker-pool-6-thread-4] SessionImpl$5.work 
INFO  2022-04-19 12:32:48,962 Connect: DONE                                                                              =>[SwingWorker-pool-6-thread-4] SessionImpl$5.work 
INFO  2022-04-19 12:32:48,963 Logging: as username fallen wings to server localhost:17171                                =>[SwingWorker-pool-6-thread-5] SessionImpl$4.work 
INFO  2022-04-19 12:32:48,986 DB: updating sets... Found new: 0                                                          =>[SwingWorker-pool-6-thread-5] SessionImpl.updateDatabase 
INFO  2022-04-19 12:32:48,991 DB: updating cards... Found new: 0                                                         =>[SwingWorker-pool-6-thread-5] SessionImpl.updateDatabase 
INFO  2022-04-19 12:32:48,992 Logging: DONE                                                                              =>[SwingWorker-pool-6-thread-5] SessionImpl$4.work 
FATAL 2022-04-19 12:32:49,019 MAGE Client UI error                                                                       =>[AWT-EventQueue-0] EDTExceptionHandler.handle 
java.lang.ExceptionInInitializerError
	at mage.client.dialog.ConnectDialog.connected(ConnectDialog.java:725)
	at mage.client.dialog.ConnectDialog.access$2100(ConnectDialog.java:35)
	at mage.client.dialog.ConnectDialog$ConnectTask.done(ConnectDialog.java:703)
	at javax.swing.SwingWorker$5.run(SwingWorker.java:737)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:832)
	at sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:842)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at mage.client.dialog.MageDialog.startModal(MageDialog.java:172)
	at mage.client.dialog.MageDialog.show(MageDialog.java:83)
	at java.awt.Component.show(Component.java:1671)
	at java.awt.Component.setVisible(Component.java:1623)
	at javax.swing.JComponent.setVisible(JComponent.java:2644)
	at mage.client.dialog.MageDialog.setVisible(MageDialog.java:89)
	at mage.client.dialog.ConnectDialog.showDialog(ConnectDialog.java:83)
	at mage.client.MageFrame.lambda$new$2(MageFrame.java:351)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at mage.deck.Standard.makeLegalSets(Standard.java:51)
	at mage.deck.Standard.<init>(Standard.java:22)
	at mage.client.util.sets.ConstructedFormats.<clinit>(ConstructedFormats.java:29)
	... 32 more
INFO  2022-04-19 12:32:59,808 Disconnecting...                                                                           =>[AWT-EventQueue-0] SessionImpl.disconnect 
INFO  2022-04-19 12:32:59,816 Disconnecting DONE                                                                         =>[AWT-EventQueue-0] SessionImpl.disconnect 
INFO  2022-04-19 12:32:59,816 Disconnected from remote task                                                              =>[AWT-EventQueue-0] MageFrame.disconnected 
INFO  2022-04-19 12:33:04,278 Connect: connecting to server localhost:17171                                              =>[SwingWorker-pool-6-thread-7] SessionImpl$5.work 
INFO  2022-04-19 12:33:04,416 Connect: DONE                                                                              =>[SwingWorker-pool-6-thread-7] SessionImpl$5.work 
INFO  2022-04-19 12:33:04,416 Logging: as username fallen to server localhost:17171                                      =>[SwingWorker-pool-6-thread-8] SessionImpl$4.work 
INFO  2022-04-19 12:33:04,421 DB: updating sets... Found new: 0                                                          =>[SwingWorker-pool-6-thread-8] SessionImpl.updateDatabase 
INFO  2022-04-19 12:33:04,423 DB: updating cards... Found new: 0                                                         =>[SwingWorker-pool-6-thread-8] SessionImpl.updateDatabase 
INFO  2022-04-19 12:33:04,424 Logging: DONE                                                                              =>[SwingWorker-pool-6-thread-8] SessionImpl$4.work 
FATAL 2022-04-19 12:33:04,450 MAGE Client UI error                                                                       =>[AWT-EventQueue-0] EDTExceptionHandler.handle 
java.lang.NoClassDefFoundError: Could not initialize class mage.client.util.sets.ConstructedFormats
	at mage.client.dialog.ConnectDialog.connected(ConnectDialog.java:725)
	at mage.client.dialog.ConnectDialog.access$2100(ConnectDialog.java:35)
	at mage.client.dialog.ConnectDialog$ConnectTask.done(ConnectDialog.java:703)
	at javax.swing.SwingWorker$5.run(SwingWorker.java:737)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:832)
	at sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
	at javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:842)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at mage.client.dialog.MageDialog.startModal(MageDialog.java:172)
	at mage.client.dialog.MageDialog.show(MageDialog.java:83)
	at java.awt.Component.show(Component.java:1671)
	at java.awt.Component.setVisible(Component.java:1623)
	at javax.swing.JComponent.setVisible(JComponent.java:2644)
	at mage.client.dialog.MageDialog.setVisible(MageDialog.java:89)
	at mage.client.dialog.ConnectDialog.showDialog(ConnectDialog.java:83)
	at mage.client.MageFrame.btnConnectActionPerformed(MageFrame.java:1065)
	at mage.client.MageFrame.access$800(MageFrame.java:83)
	at mage.client.MageFrame$7.actionPerformed(MageFrame.java:927)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:262)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

</details>

<details>
<summary>Server Log</summary>

```
INFO  2022-04-19 12:32:36,247 Starting MAGE server version 1.4.50-V2 (build: runtime)                                    =>[main] Main.main 
INFO  2022-04-19 12:32:36,249 Logging level: INFO                                                                        =>[main] Main.main 
INFO  2022-04-19 12:32:36,249 Default charset: UTF-8                                                                     =>[main] Main.main 
INFO  2022-04-19 12:32:36,251 Reading configuration from path=config\config.xml                                          =>[main] Main.main 
INFO  2022-04-19 12:32:36,342 Loading database...                                                                        =>[main] RepositoryUtil.bootstrapLocalDb 
INFO  2022-04-19 12:32:36,860 Done.                                                                                      =>[main] Main.main 
INFO  2022-04-19 12:32:36,861 Loading extension packages...                                                              =>[main] Main.main 
INFO  2022-04-19 12:32:36,861 Done.                                                                                      =>[main] Main.main 
INFO  2022-04-19 12:32:36,861 Loading cards...                                                                           =>[main] Main.main 
INFO  2022-04-19 12:32:36,889 Done.                                                                                      =>[main] Main.main 
INFO  2022-04-19 12:32:36,890 Updating user stats DB...                                                                  =>[main] Main.main 
INFO  2022-04-19 12:32:37,102 Done.                                                                                      =>[main] Main.main 
INFO  2022-04-19 12:32:37,271 Config - max seconds idle: 300                                                             =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - max game threads: 10                                                              =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - max AI opponents: 15                                                              =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - min usr name le.: 3                                                               =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - max usr name le.: 14                                                              =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - min pswrd length: 8                                                               =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - max pswrd length: 100                                                             =>[main] Main.main 
INFO  2022-04-19 12:32:37,272 Config - inv.usr name pat: [^a-z0-9_]                                                      =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - save game active: false                                                           =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - backlog size    : 200                                                             =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - lease period    : 5000                                                            =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - sock wrt timeout: 10000                                                           =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - max pool size   : 300                                                             =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - num accp.threads: 2                                                               =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - second.bind port: 17179                                                           =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - auth. activated : false                                                           =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - mailgun api key :                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - mailgun domain  :                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - mail smtp Host  :                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - mail smtpPort   :                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,273 Config - mail user       :                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,274 Config - mail passw. len.: 0                                                               =>[main] Main.main 
INFO  2022-04-19 12:32:37,274 Config - mail from addre.:                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,274 Config - google account  :                                                                 =>[main] Main.main 
INFO  2022-04-19 12:32:37,274 Loaded game types: 16, tourneys: 17, players: 4, cubes: 48, decks: 51                      =>[main] Main.main 
INFO  2022-04-19 12:32:43,481 Started MAGE server - listening on 0.0.0.0:17171/?serializationtype=java&maxPoolSize=300   =>[main] Main.main 
INFO  2022-04-19 12:32:48,969 fallen wings connected to server                                                           =>[WorkerThread#0[192.168.20.1:63852]] SessionManagerImpl.connectUser 
INFO  2022-04-19 12:32:48,986 Missing exp downloaded: 0                                                                  =>[WorkerThread#0[192.168.20.1:63852]] MageServerImpl.getMissingExpansionData 
INFO  2022-04-19 12:32:48,991 Missing cards downloaded: 0                                                                =>[WorkerThread#0[192.168.20.1:63852]] MageServerImpl.getMissingCardsData 
INFO  2022-04-19 12:32:59,811 USER REMOVE - fallen wings (Disconnected)  userId: 1bc29d81-2d6b-4605-8150-7e84737bc8c1 [] =>[CALL main-35] UserManagerImpl.lambda$null$1 
INFO  2022-04-19 12:33:04,417 fallen connected to server                                                                 =>[WorkerThread#0[192.168.20.1:63855]] SessionManagerImpl.connectUser 
INFO  2022-04-19 12:33:04,421 Missing exp downloaded: 0                                                                  =>[WorkerThread#0[192.168.20.1:63855]] MageServerImpl.getMissingExpansionData 
INFO  2022-04-19 12:33:04,423 Missing cards downloaded: 0                                                                =>[WorkerThread#0[192.168.20.1:63855]] MageServerImpl.getMissingCardsData 
```

</details>

--- a/Mage/src/main/java/mage/util/ClassScanner.java
+++ b/Mage/src/main/java/mage/util/ClassScanner.java
@@ -3,6 +3,8 @@ package mage.util;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
 import java.util.*;
 import java.util.jar.JarEntry;
@@ -43,6 +45,11 @@ public final class ClassScanner {
                         filePath = filePath.substring("file:".length(), filePath.lastIndexOf('!'));
                         jars.add(filePath);
                     } else {
+                        try {
+                            filePath = resource.toURI().getPath();
+                        } catch (URISyntaxException e) {
+                            throw new RuntimeException(e);
+                        }
                         dirs.put(filePath, packageName);
                     }
                 }

